src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/TensorPrimitives.netcore.cs

```diff
@@ -2992,6 +2992,183 @@ public static Vector512<float> Invoke(Vector512<float> x)
+            // return x.
+            // if x is NaN then raise invalid FP operation exception and return x.
+            //
+            // coshf = v/2 * exp(x - log(v)) where v = 0x1.0000e8p-1
+
+            private const uint SIGN_MASK = 0x7FFFFFFF;
+            private const uint ARG_MAX = 0x42B2D4FC;
+            private const uint LOGV = 0x3f317300;
+            private const uint HALFV = 0x3f800074;
+            private const uint INVV2 = 0x3e7ffe30;
+
+            public static float Invoke(float x) => MathF.Cosh(x);
+
+            public static Vector128<float> Invoke(Vector128<float> x)
+            {
+                Vector128<uint> ux = x.AsUInt32() & Vector128.Create(SIGN_MASK);
+                if (Vector128.GreaterThanAny(ux, Vector128.Create(ARG_MAX)))
+                {
+                    return Vector128.Create(
+                        MathF.Cosh(x.GetElement(0)),
+                        MathF.Cosh(x.GetElement(1)),
+                        MathF.Cosh(x.GetElement(2)),
+                        MathF.Cosh(x.GetElement(3)));
+                }
+
+                Vector128<float> y = ux.AsSingle();
+                Vector128<float> z = ExpOperator.Invoke(y - Vector128.Create(LOGV).AsSingle());
+                return Vector128.Create(HALFV).AsSingle() * (z + Vector128.Create(INVV2).AsSingle() * 1f / z);
+            }
+
+            public static Vector256<float> Invoke(Vector256<float> x)
+            {
+                Vector256<uint> ux = x.AsUInt32() & Vector256.Create(SIGN_MASK);
+                if (Vector256.GreaterThanAny(ux, Vector256.Create(ARG_MAX)))
+                {
+                    return Vector256.Create(Invoke(x.GetLower()), Invoke(x.GetUpper()));
+                }
+
+                Vector256<float> y = ux.AsSingle();
+                Vector256<float> z = ExpOperator.Invoke(y - Vector256.Create(LOGV).AsSingle());
+                return Vector256.Create(HALFV).AsSingle() * (z + Vector256.Create(INVV2).AsSingle() * 1f / z);
+            }
+
+#if NET8_0_OR_GREATER
+            public static Vector512<float> Invoke(Vector512<float> x)
+            {
+                Vector512<uint> ux = x.AsUInt32() & Vector512.Create(SIGN_MASK);
+                if (Vector512.GreaterThanAny(ux, Vector512.Create(ARG_MAX)))
+                {
+                    return Vector512.Create(Invoke(x.GetLower()), Invoke(x.GetUpper()));
+                }
+
+                Vector512<float> y = ux.AsSingle();
+                Vector512<float> z = ExpOperator.Invoke(y - Vector512.Create(LOGV).AsSingle());
+                return Vector512.Create(HALFV).AsSingle() * (z + Vector512.Create(INVV2).AsSingle() * 1f / z);
+            }
+#endif
+        }
+
+        /// <summary>MathF.Tanh(x)</summary>
+        private readonly struct TanhOperator : IUnaryOperator
+        {
+            // This code is based on `vrs4_tanhf` from amd/aocl-libm-ose
+            // Copyright (C) 2008-2022 Advanced Micro Devices, Inc. All rights reserved.
```

Comment:
The copyright for LogOperator below:
https://github.com/dotnet/runtime/pull/93093/files#diff-49f5fe82910e47b0f111b2a18f248ac3ee5e2d3d1b113deea0a8c87907ace9adR3176
says 2018-2019... should all of these be updated?

Reply:
Potentially? The root license for the entire repo is 2008-2022.

It looks like the per file headers are then mostly when started vs last touched within that 14 year range.